### ISSUE TITLE
do you have a simulator for archon

### ISSUE THREAD

**Issue by eh2006re** [sdss/archon#35]
Hello,  sdss-archon developers:

I am very interested in sdss-archon python package,   
I wonder if there is a simulator which we can test when we have no acceess to Archon controller hardware 

Best regards

**Comment by albireox** [comment]
Hi @eh2006re. No, we don't have an archon simulator. You can look into the `conftest.py` file in the test directory for a very simple Archon server that replies to commands being sent to it. That server helps with some of the unittesting and while it replies the same way the archon hardware does, it's not really meant as a full simulator.

**Comment by eh2006re** [comment]
HI, @albireox   Many thanks,   I will have a look. 
Regards

xiaofeng

**Comment by eh2006re** [comment]
Hello, Jose
Sorry to bother you again.
We are interested in seeing if we can use SDSS-Archon and integrate it into
our detector control system.
We are developing software on Ubuntu-20.0.4, with python3.9.  but I have
not used asyncio or docker at all.
I was not sure how to test your contest.py

I installed sdss-archon with  "pip install sdss-archon "

I also created a simple echo Archon Controller simulator and tested with a
simple client. Verified that the TCP server/Client works.
See attached.
     $ testArchonSimulatorEcho.py
     $ testArchonSimClient.py

I thought if I start testArchonSimulatorEcho.py ( we have not got the
archon controller hardware yet),
then follow the simple steps in the section on your website (see below)
I shall be able to see the TCP communication from archon to my simple
testArchonSimulatorEcho.py
[
ArchonController
<https://sdss-archon.readthedocs.io/en/latest/api.html#archon.controller.ArchonController>
provides
a mid-level interface to the Archon controller and implements the protocol
for communicating with the controller
]

However, I failed at [await archon.start()].   here is what I tested inside
python interactive session
>>>>>>>>>>>>>>>>>
~$ python
Python 3.9.7 (default, Sep 16 2021, 13:09:58)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import asyncio
>>> from archon.controller import ArchonController
>>> archon = ArchonController('my_controller', 'localhost')
>>> archon
<archon.controller.controller.ArchonController object at 0x7f4aec946310>
>>> await archon.start(False)
[ERROR]: File "<stdin>", line 1
SyntaxError: 'await' outside function
>>>>>>>>>>>>>>>>>

if I use, instead, just  archon.start(False), it returns.  But , cmd =
archon.send_command('SYSTEM')   failed.
and I never see any TCP socket communication to testArchonSimulatorEcho.py.
>>>>>>>>>>>>>>>>>>>>>>
>>>archon.start(False)
<coroutine object ArchonController.start at 0x7fa45a7267c0>
>>> cmd = archon.send_command('SYSTEM')
[ERROR]: Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File
"/home/marvel/anaconda3/lib/python3.9/site-packages/archon/controller/controller.py",
line 234, in send_command
    self.write(command.raw)
  File "/home/marvel/anaconda3/lib/python3.9/site-packages/clu/device.py",
line 119, in write
    assert self._client and self._client.writer
AssertionError
>>>>>>>>>>>>>>>>>>>>>>


Do I missed something in following the simple steps on
   https://sdss-archon.readthedocs.io/en/latest/controller.html

thanks for your help in advance

Best regards

Xiaofeng
